vue-components/src/components/LookupMenu.vue

```diff
@@ -47,8 +50,26 @@ export default Vue.extend( {
 				rootElem.style.maxHeight = menuHeight + 'px';
 			}
 		},
-	},
+		onScroll: debounce( function ( this: Vue ) {
```

Comment:
Looking good so far! I'm not quite sure why we would be declaring a parameter called `this` in this function? I don't think the lodash `debounce`  is smart enough to repass `this` as the first arg (or rather it is smart enough to understand the `this` context on it's own): https://github.com/lodash/lodash/blob/4.17.15/lodash.js#L10400-L10423.

Reply:
As far as I understand it, `this` as it is written here is not actually a function parameter, but a means of letting TypeScript know that you're sure it's bound to a Vue object when it's called.

https://www.typescriptlang.org/docs/handbook/functions.html#this-parameters

Reply:
That is well confusing... but hey, typescript ;) guess I never noticed it before as I use arrows pretty excessively...

Reply:
If the sole purpose of passing `this` is to use the `$emit` function (see my other comment below), one can do that with [vm](https://vuejs.org/v2/api/#vm-emit).
I think it's less awkward to pass `vm` to `debounce` as it's clearer what it is (also searchable in the vue docs).

Reply:
could you be more clear? as in code example. thanks. :)

Reply:
After revisiting the docs, we can't use vm. For some reason I thought `vm` always exists and is the same as the `this` of the vue instance, but that is wrong. So scratch that idea.